scrape/scrape.go

```diff
@@ -1451,6 +1461,7 @@ loop:
 		var (
 			et          textparse.Entry
 			sampleAdded bool
+			hasMetadata bool
```

Comment:
nit: hasMetadata seems not totally accurate, since metadata can exist but still not be written. `shouldWriteMetadata`?